darts/utils/data/shifted_dataset.py

```diff
@@ -0,0 +1,114 @@
+from typing import Union, Sequence, Optional
+from ...timeseries import TimeSeries
+from .timeseries_dataset import TimeSeriesDataset
+from ..utils import raise_if_not
+
+
+class ShiftedDataset(TimeSeriesDataset):
+    def __init__(self,
+                 input_series: Union[TimeSeries, Sequence[TimeSeries]],
+                 target_series: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+                 seq_length: int = 12,
+                 shift: int = 1,
+                 max_samples_per_ts: Optional[int] = None):
+        """
+        A time series dataset containing tuples of (input, target) series, both of length `seq_length`.
+        The "target" series is in advance of `shift` time steps compared to the "input" series.
+
+        The input and target time series are sliced together, and therefore must have the same time axes.
```

Comment:
To me it's not 100% clear what 'sliced together' means. Could there be a better way of saying this? Maybe "`input_series` and `target_series` must have the same time indices' would suffice?